test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpExporterTest.cs

```diff
@@ -322,6 +326,51 @@ public void UseOpenTelemetryProtocolActivityExporterWithCustomActivityProcessor(
             Assert.True(endCalled);
         }
 
+        [Theory]
+        [InlineData("key=value", new string[] { "key" }, new string[] { "value" })]
+        [InlineData("key1=value1,key2=value2", new string[] { "key1", "key2" }, new string[] { "value1", "value2" })]
+        [InlineData("key1 = value1, key2=value2 ", new string[] { "key1", "key2" }, new string[] { "value1", "value2" })]
+        public void GetMetadataFromHeadersWorksCorrectFormat(string headers, string[] keys, string[] values)
+        {
+            var otlpExporter = new OtlpTraceExporter(new OtlpExporterOptions());
+            var metadata = typeof(OtlpTraceExporter)
+                    .GetMethod("GetMetadataFromHeaders", BindingFlags.NonPublic | BindingFlags.Static)
+                    .Invoke(otlpExporter, new object[] { headers }) as Metadata;
+
+            Assert.Equal(keys.Length, metadata.Count);
+
+            for (int i = 0; i < keys.Length; i++)
+            {
+                Assert.Contains(metadata, entry => entry.Key == keys[i] && entry.Value == values[i]);
+            }
+        }
+
+        [Theory]
+        [InlineData("headers")]
+        [InlineData("key,value")]
+        [InlineData("key==value")]
+        [InlineData("key1=value1;key2=value2")] // semicolon separated values are not supported (https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter.md#specifying-headers-via-environment-variables)
+        public void GetMetadataFromHeadersThrowsExceptionOnOnvalidFormat(string headers)
+        {
+            var otlpExporter = new OtlpTraceExporter(new OtlpExporterOptions());
+            try
+            {
+                var metadata = typeof(OtlpTraceExporter)
+                            .GetMethod("GetMetadataFromHeaders", BindingFlags.NonPublic | BindingFlags.Static)
+                            .Invoke(otlpExporter, new object[] { headers }) as Metadata;
+            }
+            catch (Exception ex)
+            {
+                // The exception thrown here is System.Reflection.TargetInvocationException. The exception thrown by the method GetMetadataFromHeaders
+                // is captured in the inner exception
+                Assert.IsType<ArgumentException>(ex.InnerException);
```

Comment:
What's the relation between `ArgumentException` and the `System.Reflection.TargetInvocationException` type in the comment?

Reply:
Since the method that we are testing is private and we are using reflection to invoke it with invalid arguments, when it throws an exception and it's caught as a `System.Reflection.TargetInvocationException` as that's what the methods invoked through reflection throw. The original exception, in this case, `ArgumentException` which caused the current exception is captured as InnerExcpetion of the current excpetion.

Reply:
you can make the method internal so they can be called directly from unit tests.